WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsViewModel.kt

```diff
@@ -0,0 +1,252 @@
+package org.wordpress.android.ui.sitecreation.verticals
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.support.annotation.StringRes
+import kotlinx.coroutines.experimental.CoroutineScope
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.delay
+import kotlinx.coroutines.experimental.launch
+import kotlinx.coroutines.experimental.withContext
+import org.apache.commons.lang3.StringUtils
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.vertical.VerticalModel
+import org.wordpress.android.fluxc.store.VerticalStore.OnVerticalsFetched
+import org.wordpress.android.models.networkresource.ListState
+import org.wordpress.android.models.networkresource.ListState.Error
+import org.wordpress.android.models.networkresource.ListState.Loading
+import org.wordpress.android.modules.IO_DISPATCHER
+import org.wordpress.android.modules.MAIN_DISPATCHER
+import org.wordpress.android.ui.sitecreation.usecases.DummyOnVerticalsHeaderInfoFetched
+import org.wordpress.android.ui.sitecreation.usecases.DummyVerticalsHeaderInfoModel
+import org.wordpress.android.ui.sitecreation.usecases.FetchVerticalsHeaderInfoUseCase
+import org.wordpress.android.ui.sitecreation.usecases.FetchVerticalsUseCase
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsViewModel.VerticalsListItemUiState.VerticalsFetchSuggestionsErrorUiState
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsViewModel.VerticalsListItemUiState.VerticalsModelUiState
+import org.wordpress.android.viewmodel.SingleLiveEvent
+import javax.inject.Inject
+import javax.inject.Named
+import kotlin.coroutines.experimental.CoroutineContext
+
+private const val throttleDelay: Int = 500
+
+class NewSiteCreationVerticalsViewModel @Inject constructor(
+    private val dispatcher: Dispatcher,
+    private val fetchVerticalsHeaderInfoUseCase: FetchVerticalsHeaderInfoUseCase,
+    private val fetchVerticalsUseCase: FetchVerticalsUseCase,
+    @Named(IO_DISPATCHER) private val IO: CoroutineContext,
+    @Named(MAIN_DISPATCHER) private val MAIN: CoroutineContext
+) : ViewModel(), CoroutineScope {
+    private val job = Job()
+    private var fetchVerticalsJob: Job? = null
```

Comment:
I think this is where my confusion with `FetchVerticalsUseCase` comes from. I was expecting the cancellation and the delay to be in the use case instead. I am not saying that'd be better, because I am not sure yet.